pkg/domain/infra/abi/network.go

```diff
@@ -12,10 +13,33 @@ import (
 )
 
 func (ic *ContainerEngine) NetworkList(ctx context.Context, options entities.NetworkListOptions) ([]types.Network, error) {
+	// dangling filter is not provided by netutil
+	var wantDangling bool
+
+	val, filterDangling := options.Filters["dangling"]
+	if filterDangling {
+		var err error
+
+		wantDangling, err = strconv.ParseBool(val[0])
```

Comment:
however, being "dangling" a binary filter, I wonder if a good user experience would be to just error out if more than one value is set (or none).
Also to avoid to decide what to do when two valid, but opposite values are passed (like in your first example)

Reply:
good point, I just checked
```bash
 ❯ docker network ls --filter dangling=true --filter dangling=false
Error response from daemon: got more than one value for filter key "dangling"
```